### DISCUSSION TITLE
Command Line Interface?

### DISCUSSION THREAD

**Discussion by Orcomp** [SeekStorm/SeekStorm#2] [General]
Would it be possible to provide a CLI tool that would allow us to:

- Interact with the library from the command line on Windows, Linux, MaxOS
- Specify a JSON schema
- Point to a directory to create an index
- Query an index for results (with multiple output formats. i.e. STDOUT, JSON etc...)
- Manage the index (add, remove items)

Easy installation and updates through scoop.sh would be beneficial.

(Thank you for building this library it looks amazing.)

**Comment by wolfgarbe** [comment]
Most of what you want can be achieved already today with the SeekStorm server. It offers a REST API that can be accessed via cURL from the command line: https://github.com/SeekStorm/SeekStorm/tree/main/src/seekstorm_server 

What is still missing is connecting to STDIN/STDOUT. We have implemented a very rudimentary CLI wrapper that uses STDIN/STDOUT for SeekStorm for benchmarking:
https://github.com/SeekStorm/search-benchmark-game/blob/master/engines/seekstorm-0.1.15-singlefield/src/bin/build_index.rs
https://github.com/SeekStorm/search-benchmark-game/blob/master/engines/seekstorm-0.1.15-singlefield/src/bin/do_query.rs

So yes, implementing a command line interface for SeekStorm is possible.

**Reply by Orcomp** [reply]
Thanks. Will take a closer look at the server.
